### PR TITLE
Add studio participation

### PR DESCRIPTION
**Pull request by jianoway**
## Before executing any command

Screenshot 2021-10-09 at 3 15 28 AM

## After running ```part 1, 2 ses/1 add/4```

Screenshot 2021-10-09 at 3 17 22 AM

The range for participation count I've set is from 0 to 100, with the Avenger being able to modify this participation count with any integer value from -100 to 100 via the command. As follows:

## After running  ```part 1 ses/1 add/ 100```

Screenshot 2021-10-09 at 3 17 57 AM

As you can see, the participation count caps at 100. All values are arbitrary and subject to change if necessary.

Note how the even though the command parsed 2 steps earlier affected both students and executed without bugging, only student 1's participation is reflected as student 2 has been indicated as not having attended the session. Thus, their attendance and participation count are omitted.

To view student 2's participation, we can do as follows:

## After running ```attendance 2 ses/1 att/1```

Screenshot 2021-10-09 at 3 18 38 AM

We can also negatively modify the student's participation count as follows:

## After running ```part 1, 2 ses/1 add/-40

Screenshot 2021-10-09 at 3 19 05 AM

As above, I've restricted the range for participation count to not fall below 0, with any integer value beyond 0 defaulting to 0 and any integer value above 100 defaulting to 100 respectively.

If the tutor passes in an invalid participation count e.g 101, the error message pops ups as follows:

## After running ```part 1, 2 ses/1 add/101```

Screenshot 2021-10-09 at 3 30 44 AM

Please do let me know if you have any suggestions, thanks!

### CODE REVIEW

**Comment by phuccuongngo99** [comment]
After running `attendance 2 ses/1 att/1`, it jumps to `[1: 4]`. Is the default 4 participation mark?

**Comment by phuccuongngo99** [comment]
LGTM! Very comprehensive test !

File: src/test/java/seedu/academydirectory/logic/commands/ParticipationCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        // different types -> returns false
+        assertFalse(standardCommand.equals(new ClearCommand()));
+
+        // different indexArrayList
+        assertFalse(standardCommand.equals(
+                new ParticipationCommand(12, STUDIO_SESSION_STUB, INDEX_ARRAYLIST_STUB_2)));
+
+        // different participation
+        assertFalse(standardCommand.equals(
+                new ParticipationCommand(11, STUDIO_SESSION_STUB, INDEX_ARRAYLIST_STUB)));
```
Comment (phuccuongngo99) [inline src/test/java/seedu/academydirectory/logic/commands/ParticipationCommandTest.java:118]: Comprehensive test!

**Review by phuccuongngo99 — DISMISSED** [review]
LGTM!

**Comment by jianoway** [comment]
> After running `attendance 2 ses/1 att/1`, it jumps to `[1: 4]`. Is the default 4 participation mark?

Ah no it isn't.  By default it's 0. In the previous step I ran part 1, 2 ses/1 add/4 which added participation score of 4 to the student.

**Comment by phuccuongngo99** [comment]
> Ah no it isn't. By default it's 0. In the previous step I ran part 1, 2 ses/1 add/4 which added participation score of 4 to the student.

I think it's a bit confusing for users.
Maybe this is more intuitive?
- if the tutor add participation, it will mark attendance as true and update that participation
- if the tutor decrease participation to 0, the attendance can still be there

This is because some students may attend, but don't participate in anything

File: src/test/java/seedu/academydirectory/model/student/ParticipationTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+        participation.add(1, 100);
+        assertTrue(participation.getParticipationArray()[0] == 100);
+
+        participation.add(1, 500);
+        assertTrue(participation.getParticipationArray()[0] == 500);
+
+        participation.add(1, 500);
+        assertTrue(participation.getParticipationArray()[0] == 500);
+    }
```
Comment (phuccuongngo99) [inline src/test/java/seedu/academydirectory/model/student/ParticipationTest.java:41]: Comprehensive test

**Pull request merged**